flink-tests/src/test/java/org/apache/flink/test/checkpointing/UnalignedCheckpointTestBase.java

```diff
@@ -871,13 +875,15 @@ public String toString() {
         private final LongCounter lostCounter = new LongCounter();
         private final LongCounter duplicatesCounter = new LongCounter();
         private final IntCounter numFailures = new IntCounter();
+        private final Duration backpressureInterval;
         private ListState<State> stateList;
         protected transient State state;
         protected final long minCheckpoints;
-        protected boolean backpressure;
+        @Nullable private Deadline backpressureUntil;
 
-        protected VerifyingSinkBase(long minCheckpoints) {
+        protected VerifyingSinkBase(long minCheckpoints, long checkpointingInterval) {
             this.minCheckpoints = minCheckpoints;
+            this.backpressureInterval = Duration.ofMillis(checkpointingInterval);
```

Comment:
I think in this commit you are changing too many independent things? I'm not following those back pressure changes, why do we need them?

Also how is it supposed to be working? We are backpressuring only for the first checkpoint? Or are you bumping this backpressure after every snapshot state call? But in that case, I would be afraid backpressure is gone until we trigger next checkpoint, if there is even a small delay in for example checkpoint notification.

Reply:
Quite possible, but it's actually covering more this way (some tasks may have data, other tasks don't) whereas we used to test only (every task has data). It depends a bit on the run; so kind of randomized.

If you'd rather have more deterministic behavior, we could also set backpressureInterval to `2 * checkpointingInterval`. Still a chance for backpressure to be gone, but much rarer.